### PR TITLE
Holoparasite Shield Ability (+ minor rebalancing)

### PR DESCRIPTION
**Pull request by Absolucy**
This adds a new ability, Shield, based off of [20th Century Boy](https://jojo.fandom.com/wiki/20th_Century_Boy). When activated, you'll have _nigh-absolute invulnerability_, however you will be completely immobile and unable to use any items (enforced using `TRAIT_NOINTERACT`). The shield can only last so long (15 seconds on Potential F, 5 minutes on Potential A), and while you can't take any damage while inside it, that won't stop people from camping right outside waiting for you to exit it, or RCDing you in with walls, etc etc etc.

Also did some rebalancing - many potential-affected stats now have an actual table of values based on potential rather than some random asspulled equation.

Also some code restyling/refactoring.

# Changelog

:cl: Lucy. sprites by UselessTheremin
add: Added a new holoparsite ability, Shield!
add: The Shield ability allows a holoparasite to project an impenetrable shield around you, however it will not last forever (from potential F to A: 15 seconds, 30 seconds, 1 minute, 2 minutes, 5 minutes), and doing so will render you completely immobile and unable to use items. Let's hope nobody is waiting outside your shield waiting to catch you when you leave it!
tweak: Explosive holoparasites and surveillance snare holoparasites now have action buttons.
tweak: Rebalanced how long explosive holopara bombs last (from potential F to A: 15 seconds, 25 seconds, 40 seconds, 90 seconds, 3 minutes)
tweak: The speedup from a Frenzy holoparasite is now affected by potential.
tweak: Pocket Dimension now shows up as scrambled on suit sensors
/:cl:

### CODE REVIEW

**Review by nesisknesis — APPROVED** [review]

**Comment by JamieD1** [comment]
Woah woah woah, why are we suddenly changing to the wrong spelling of colour.

File: yogstation/code/modules/guardian/abilities/major/shield.dm [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+/datum/guardian_ability/major/shield/Remove()
+	. = ..()
+	QDEL_LIST(shields)
+
+/datum/guardian_ability/major/shield/Recall()
+	QDEL_LIST(shields)
+
+/obj/effect/proc_holder/spell/self/guardian_shield
+	name = "Activate Shield"
+	desc = "Create an inpenetrable shield around you and your master."
```
Comment (Hopekz) [inline yogstation/code/modules/guardian/abilities/major/shield.dm]:
```suggestion
	desc = "Create an impenetrable shield around you and your master."
```

File: yogstation/code/modules/guardian/abilities/minor/snares.dm [hunk 2]
```diff
@@ -2,36 +2,42 @@
...
-	set category = "Guardian"
-	set desc = "Set an invisible snare that will alert you when living creatures walk over it. Max of 5"
-	if(snares.len <6)
-		var/turf/snare_loc = get_turf(src.loc)
+	. = ..()
+	QDEL_LIST(snares)
+
+/datum/action/guardian/arm_snare
+	name = "Arm Snare"
+	desc = "Set an invisible snare that will alert you when living creatures walk over it. Max of 5"
```
Comment (Hopekz) [inline yogstation/code/modules/guardian/abilities/minor/snares.dm]:
```suggestion
	desc = "Set an invisible snare that will alert you when living creatures walk over it. Max of 5."
```

**Comment by Hopekz** [comment]
> colour

colour and color don't even do the same thing in the code.

colour only exists because color is a restricted variable name since it is already used natively by all objects.

**Comment by Hopekz** [comment]
> Woah woah woah, why are we suddenly changing to the wrong spelling of colour.

colour and color don't even do the same thing in the code.

colour only exists because color is a restricted variable name since it is already used natively by all objects.

**Comment by Absolucy** [comment]
> colour only exists because color is a restricted variable name since it is already used natively by all objects.

@Hopekz it's fine for a /datum

**Pull request closed without merging**